src/hyperlight_host/src/func/guest_err.rs

```diff
@@ -14,31 +14,39 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-use hyperlight_common::flatbuffer_wrappers::guest_error::ErrorCode;
+use hyperlight_common::flatbuffer_wrappers::guest_error::{ErrorCode, GuestError as FbGuestError};
 
 use crate::error::HyperlightError::{GuestError, OutBHandlingError, StackOverflow};
+use crate::mem::mgr::SandboxMemoryManager;
 use crate::mem::shared_mem::HostSharedMemory;
 use crate::metrics::{METRIC_GUEST_ERROR, METRIC_GUEST_ERROR_LABEL_CODE};
-use crate::sandbox::mem_mgr::MemMgrWrapper;
 use crate::{log_then_return, Result};
+
 /// Check for a guest error and return an `Err` if one was found,
 /// and `Ok` if one was not found.
-pub(crate) fn check_for_guest_error(mgr: &MemMgrWrapper<HostSharedMemory>) -> Result<()> {
-    let guest_err = mgr.as_ref().get_guest_error()?;
+pub(crate) fn check_for_guest_error(
+    mgr: &mut SandboxMemoryManager<HostSharedMemory>,
+) -> Result<()> {
+    let peb = mgr.memory_sections.read_hyperlight_peb()?;
+    let (odr_buffer, odr_size) = peb.get_output_data_guest_region()?;
+    let maybe_guest_err = mgr
+        .shared_mem
+        .try_pop_buffer_into::<FbGuestError>(odr_buffer as usize, odr_size as usize);
+
+    let guest_err = if let Ok(err) = maybe_guest_err {
```

Comment:
Here we circumvent our shared-memory abstraction again by reading the raw PDB from memory, which we should probably not do. In addition, I'm not a big fan of the way we check for errors here by trying to deserializng an error and see if it works or not